core/whistle_number_manager-1.0.0/src/wh_number_manager.erl

```diff
@@ -1007,7 +1007,10 @@ prepare_find_results([Number|Numbers], ModuleName, ModuleResults, Found, Opts) -
                       | Found]
              end,
     case catch wnm_number:get(Number) of
-        #number{state=State} ->
+        #number{state=State
+                ,module_data=JObj
+                ,module_name = ModuleName
```

Comment:
@k-anderson  if it matches this clause it will not be save, ie, if its the same module and the same module_data (JObj) then we don't save it (no need to).

another clause was added to save it in case there is a difference